libkmod/libkmod.h

```diff
@@ -689,29 +674,50 @@ enum kmod_insert {
 int kmod_module_insert_module(struct kmod_module *mod, unsigned int flags,
 							const char *options);
 
+/**
+ * kmod_probe:
+ * @KMOD_PROBE_FORCE_VERMAGIC: ignore kernel version magic
+ * @KMOD_PROBE_FORCE_MODVERSION: ignore symbol version hashes
+ * @KMOD_PROBE_IGNORE_COMMAND: ignore install commands and softdeps configured
+ * in the system
+ * @KMOD_PROBE_IGNORE_LOADED: do not check whether the module is already
+ * live in kernel or not
+ * @KMOD_PROBE_DRY_RUN: dry run, do not insert module, just call the
+ * associated callback function
+ * @KMOD_PROBE_FAIL_ON_LOADED: probe will fail if KMOD_PROBE_IGNORE_LOADED is
+ * not specified and the module is already live in kernel
+ * @KMOD_PROBE_APPLY_BLACKLIST_ALL: apply KMOD_FILTER_BLACKLIST filter to this
+ * module and its dependencies. If any of the dependencies (or the module) is
+ * blacklisted, the probe will fail, unless the blacklisted module is already
+ * live in kernel
+ * @KMOD_PROBE_APPLY_BLACKLIST: probe will fail if the module is blacklisted
+ * @KMOD_PROBE_APPLY_BLACKLIST_ALIAS_ONLY: probe will fail if the module is an
+ * alias and is blacklisted
+ *
+ * Bitmask which defines the behaviour of kmod_module_probe_insert_module().
+ */
+enum kmod_probe {
+	KMOD_PROBE_FORCE_VERMAGIC =		0x00001,
+	KMOD_PROBE_FORCE_MODVERSION =		0x00002,
+	KMOD_PROBE_IGNORE_COMMAND =		0x00004,
+	KMOD_PROBE_IGNORE_LOADED =		0x00008,
+	KMOD_PROBE_DRY_RUN =			0x00010,
+	KMOD_PROBE_FAIL_ON_LOADED =		0x00020,
+
+	/*
+	 * codes below can be used in return value, too
+	 * XXX: emil: return value of what, document this
```

Comment:
"Flags to kmod_module_probe_insert_module" and the KMOD_PROBE_APPLY_BLACKLIST_*, besides being input as flag can also be a return value if the function is failing because of that flag.

Reply:
that is verified in the function near the beginning of the function in the ugly assignment + check for these flags.